frontend/src/metabase/dashboard/actions/data-fetching-typed.ts

```diff
@@ -203,10 +239,341 @@ export const markCardAsSlow = createAction(MARK_CARD_AS_SLOW, (card: Card) => {
+            ),
+            ignore_cache: ignoreCache,
+          },
+          queryOptions,
+        ),
+      );
+    } else if (dashboardType === "transient" || dashboardType === "inline") {
+      result = await fetchDataOrError(
+        maybeUsePivotEndpoint(MetabaseApi.dataset, card)(
+          { ...datasetQuery, ignore_cache: ignoreCache },
+          queryOptions,
+        ),
+      );
+    } else {
+      const dashcardBeforeEditing = getDashCardBeforeEditing(
+        getState(),
+        dashcard.id,
+      );
+      const hasReplacedCard =
+        dashcard.card_id != null &&
+        dashcardBeforeEditing &&
+        dashcardBeforeEditing.card_id !== dashcard.card_id;
+
+      const shouldUseCardQueryEndpoint =
+        isNewDashcard(dashcard) ||
+        (isQuestionDashCard(dashcard) &&
+          isNewAdditionalSeriesCard(card, dashcard)) ||
+        hasReplacedCard;
+
+      // new dashcards and new additional series cards aren't yet saved to the dashboard, so they need to be run using the card query endpoint
+      const endpoint = shouldUseCardQueryEndpoint
+        ? CardApi.query
+        : DashboardApi.cardQuery;
+
+      const requestBody = shouldUseCardQueryEndpoint
+        ? { cardId: card.id, ignore_cache: ignoreCache }
+        : {
+            dashboardId: dashcard.dashboard_id,
+            dashcardId: dashcard.id,
+            cardId: card.id,
+            parameters: datasetQuery.parameters,
+            ignore_cache: ignoreCache,
+            dashboard_id: dashcard.dashboard_id,
+            dashboard_load_id: dashboardLoadId,
+          };
+
+      result = await fetchDataOrError(
+        maybeUsePivotEndpoint(endpoint, card)(requestBody, queryOptions),
+      );
+    }
+
+    setFetchCardDataCancel(card.id, dashcard.id, null);
+    clearTimeout(slowCardTimer);
+
+    return {
+      dashcard_id: dashcard.id,
+      card_id: card.id,
+      result: cancelled ? null : result,
+      currentTime: performance.now(),
+    };
+  },
+);
+
+export const fetchCardData =
+  (
+    card: Card,
+    dashcard: DashboardCard,
+    options: {
+      reload?: boolean;
+      clearCache?: boolean;
+      ignoreCache?: boolean;
+      dashboardLoadId?: string;
+    } = {},
+  ) =>
+  (dispatch: Dispatch) => {
+    dispatch(fetchCardDataAction({ card, dashcard, options }));
+  };
+
+fetchCardData.fulfilled = fetchCardDataAction.fulfilled;
+
+export const CLEAR_CARD_DATA = "metabase/dashboard/CLEAR_CARD_DATA";
+
+export const clearCardData = createAction(
+  CLEAR_CARD_DATA,
+  (cardId, dashcardId) => ({ payload: { cardId, dashcardId } }),
+);
+
+function getDatasetQueryParams(datasetQuery: DatasetQuery) {
+  const type = datasetQuery.type;
+  if (type === "native") {
+    return {
+      type,
+      native: datasetQuery.native,
+      parameters: datasetQuery.parameters
+        ?.map(parameter => ({
+          ...parameter,
+          value: parameter.value ?? null,
+        }))
+        .sort(sortById),
```

Comment:
I wonder why parameters are only specified in native queries here

Reply:
<img width="615" alt="image" src="https://github.com/user-attachments/assets/8f4c9796-2c53-44e4-bd0b-41a1ffeaf22f">
yep, only native should have them. If you take out the if-statements her the error is "parameters doesn't exist in StructuredDatasetQuery" or something like that